### PR TITLE
Switch TPv2 E2E tests to Cypress

### PR DESCRIPTION
**Pull request by ocket8888**
This PR switches the TPv2 E2E testing suite from NightwatchJS to Cypress.

"Why?", you may ask. Because it took me all of two days to replicate all of our test suite in Cypress - that's how easy it is to use. For comparison: _I have never even once been able to get the Nightwatch tests to run on my machine_. With Cypress, the typings are provided by the library itself, testing on Chrome and Firefox took exactly zero time and effort to figure out, and the tests actually just do the things I want them to do without me having to spend countless hours needling them with the correct set of `await`s and chained commands. Which is why most all of our tests do little more than check for the presence of some DOM elements. Maybe with Cypress, we could fix that.

tl;dr: Nightwatch is extremely hard to use and Cypress is extremely easy.

<hr/>

## Which Traffic Control components are affected by this PR?
- Traffic Portal (experimental v2)

## What is the best way to verify this PR?
Run the end-to-end tests. Which doesn't take very long and can be depended upon to actually work.

## PR submission checklist
- [x] This PR has tests
- [x] This PR has documentation
- [x] This PR doesn't need a CHANGELOG.md entry
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY**

### CODE REVIEW

File: experimental/traffic-portal/cypress.config.ts
Comment (shamrickus) [inline experimental/traffic-portal/cypress.config.ts:1]: I think it would make more sense for this to be in the `cypress` folder, the root folder is already pretty crowded.

File: .github/actions/tpv2-integration-tests/entrypoint.sh
Comment (shamrickus) [inline .github/actions/tpv2-integration-tests/entrypoint.sh:1]: Not sure if it's related to your changes but when I download the GHA artifact for the end-to-end tests, the TP log shows an error `Error: Cannot find module '/home/runner/work/trafficcontrol/trafficcontrol/experimental/traffic-portal/dist/traffic-portal/server/main.js'`

File: experimental/traffic-portal/cypress.config.ts
Comment (ocket8888) [reply experimental/traffic-portal/cypress.config.ts:1]: I'd rather not touch it, just because that's where the schematic put it when I `ng add`ed Cypress

File: .github/actions/tpv2-integration-tests/entrypoint.sh
Comment (ocket8888) [reply .github/actions/tpv2-integration-tests/entrypoint.sh:1]: Hm. Could be, I'll look into it

Comment (ocket8888) [reply .github/actions/tpv2-integration-tests/entrypoint.sh:1]: ah, yeah, it's still trying to run the server. Curious that doesn't crash the script, but whatever. There shouldn't need to be any TPv2 server logs because it no longer needs to run the TPv2 server.

**Review by shamrickus — APPROVED** [review]

**Comment by zrhoffman** [comment]
This is making the TPv2 RPM fail to build

From the PR of #7922:

> npm ERR! Cypress cannot write to the cache directory due to file permissions

**Comment by ocket8888** [comment]
that's interesting. It shouldn't be attempting to compile or run the e2e tests to build the RPM.

**Comment by zrhoffman** [comment]
Reverted due to breaking tests in 4b6af898d8, please re-PR with tests fixed

**Comment by ocket8888** [comment]
Reverting something experimental because it affects the ability to build distributables for something that isn't distributed (because it's experimental) seems a bit extreme.

**Comment by zrhoffman** [comment]
If we don't want it to affect the default builds, then

- It should not be built when someone just runs `./pkg` without specifying optional builds with `-o`, and
- It should not be included in the *CDN-in-a-Box CI* GHA workflow.

But because it is included in `./pkg` by default and the *CDN-in-a-Box CI* GHA workflow, it broke tests for completely unrelated components, and that's why it was reverted.
